lambdas/user_access_patterns/lambda_for_createUser/index.js

```diff
@@ -1,6 +1,58 @@
-const AWS = require('aws-sdk');
-exports.handler = async (event) => {
-    // TODO implement
-    const response = (success, reason, data) => { return { success: success, reason: reason, data: data } }
-    return response(true, null, null);
+const AWS = require("aws-sdk");
+const crypto = require("crypto");
+//AWS.config.loadFromPath("../../../keys.json");
+//Use the below code for local setup
+AWS.config.update({
+  region: "local",
+  endpoint: "http://localhost:8000",
+});
+
+const dynamo = new AWS.DynamoDB.DocumentClient();
+function response(statusCode, message) {
+  return {
+    statusCode: statusCode,
+    body: JSON.stringify(message),
+  };
+}
+
+function hmac_hash(secretKey, password) {
+  const hashingSecret = secretKey;
+  const plainText = password;
+  const hashedStr = crypto
+    .createHmac("sha256", hashingSecret)
+    .update(plainText)
+    .digest("hex");
+
+  return hashedStr;
+}
+
+module.exports.createUser = (event, ctx, callback) => {
```

Comment:
1. what's the difference between `module.exports.createUser` and `exports.createUser`?
2. Also if you are not using `ctx` then when is its relevance in the function signature?
3. Why not the function in asynchronous?

Reply:
2. For using callback
3. I like working with promise

Reply:
it's not about liking, use async functions only.

Reply:
@lakshya-20 and the 1st point.

Reply:
use async function only.

Reply:
1. `module.exports` and `exports` both point at the same object. So you can add `createUser` to this object using either of two, it doesn't matter which you use in this case.

Reply:
so basically they are same?